docs/user/rstudio/ide/guide/productivity/project-templates.qmd

```diff
@@ -0,0 +1,179 @@
+
+## Example
+
+We'll use the [ptexamples](https://github.com/rstudio/ptexamples) package to illustrate how a project template can be defined. This package exports a project template that is presented like so from the **New Project...** wizard:
+
+![](images/ptexamples-wizard.png){fig-alt="A screenshot of the Create Example Project Template wizard with a directory name and other user input options."}
+
+After the user clicks **Create Project**, a new project will be created, and the `hello_world()` template function will be called to initialize the project. Here's what the IDE looks like after calling this template function with the aforementioned inputs:
+
+![](images/ptexamples-rstudio.png){fig-alt="A screenshot of the newly created ptexamples RStudio project displaying the index with the user inputs."}
+
+To try using this project template locally, install the package:
+
+``` r
+# if needed install.packages("devtools")
+devtools::install_github("rstudio/ptexamples")
+```
+
+Restart RStudio and then access the `ptexamples` template from the **New Project** wizard.
+
+## Defining the Template Function
+
+The `ptexamples` package contains a function, [`hello_world()`](https://github.com/rstudio/ptexamples/blob/master/R/hello_world.R), intended to be used as a project template. The function has the signature:
+
+``` r
+hello_world <- function(path, ...) {
+  # ensure path exists
+  dir.create(path, recursive = TRUE, showWarnings = FALSE)
+
+  # generate header for file
+  header <- c(
+    "# This file was generated by a call to 'ptexamples::hello_world()'.",
+    "# The following inputs were received:",
+    ""
+  )
+
+  # collect inputs and paste together as 'Parameter: Value'
+  dots <- list(...)
+  text <- lapply(seq_along(dots), function(i) {
+    key <- names(dots)[[i]]
+    val <- dots[[i]]
+    paste0(key, ": ", val)
+  })
+
+  # collect into single text string
+  contents <- paste(
+    paste(header, collapse = "\n"),
+    paste(text, collapse = "\n"),
+    sep = "\n"
+  )
+
+  # write to index file
+  writeLines(contents, con = file.path(path, "INDEX"))
+}
+```
+
+When this function is invoked by RStudio, it will receive the `path` as the path to the newly created project, as well as arguments as received from its input widgets. This function collects its inputs, and echos them to a file called `INDEX` in the newly-created project path.
+
+## Defining the Template Metadata
+
+After defining a function to be used as a project template, some initial metadata must be provided to let RStudio know how to use this function as a project template. The `ptexamples` package defines a project template in a file located at:
+
+<https://github.com/rstudio/ptexamples/blob/master/inst/rstudio/templates/project/hello_world.dcf>
+
```

Comment:
on my mac screen this link extends into the page's table of contents and looks bad. Maybe shorten the link text to just the file path without all the github specific stuff: [inst/rstudio/templates/project/hello_world.dcf](https://github.com/rstudio/ptexamples/blob/master/inst/rstudio/templates/project/hello_world.dcf) ?
<img width="1007" alt="image" src="https://user-images.githubusercontent.com/114421526/196551828-c7885829-a2c8-4909-93ac-2dd23716e5c3.png">

Reply:
Made it a traditional `[inst/rstudio/templates/project/hello_world.dcf](https://github.com/rstudio/ptexamples/blob/master/inst/rstudio/templates/project/hello_world.dcf)` link rather than the raw <Github> link.